core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -674,10 +674,11 @@ export class LoggedInUser extends BaseUser {
       throw new Error('The Watch A Video button does not exist!');
     }
     await Promise.all([
-      this.page.waitForNavigation(),
+      this.page.waitForNavigation({waitUntil: ['load', 'networkidle2']}),
       this.clickOn(watchAVideoButton),
     ]);
 
+    await this.waitForText('Create New Account');
     const url = this.getCurrentUrlWithoutParameters();
```

Comment:
Occasionally, the URL was being fetched before the page had fully loaded, leading to differences with the expected URL. Therefore, made changes to ensure the page is fully loaded before the URL is fetched.

Reply:
Can you explain what the URL is in the intermediate stage and what it is after the page settles down?